modules/core/localizer.js

```diff
@@ -348,13 +350,24 @@ export function coreLocalizer() {
 
     // Returns the localized text wrapped in an HTML element encoding the locale info
     localizer.t.html = function(stringId, replacements, locale) {
-        const info = localizer.tInfo(stringId, replacements, locale);
-        // text may be empty or undefined if `replacements.default` is
-        return info.text ? localizer.htmlForLocalizedText(info.text, info.locale) : '';
-    };
+      // replacement string might be html unsafe, so we need to escape it except if it is explicitly marked as html code
+      replacements = Object.assign({}, replacements);
+      for (var k in replacements) {
+        if (typeof replacements[k] === 'string') {
+          replacements[k] = escape(replacements[k]);
+        }
+        if (typeof replacements[k] === 'object' && typeof replacements[k].html === 'string') {
+          replacements[k] = replacements[k].html;
+        }
+      }
 
-    localizer.htmlForLocalizedText = function(text, localeCode) {
-        return `<span class="localized-text" lang="${localeCode || 'unknown'}">${text}</span>`;
```

Comment:
I think the intention in #7998 was to allow for each localizable string to be marked up with its language, even if it’s a string that gets inserted into another string as a replacement: #7963. That’s a reasonable idea, though certainly _much_ less important than string sanitization. We should reopen that issue to track a more secure way of handling mixed-language strings. It might require an alternative method that returns both the localized text and the resolved language (a “tagged string”, as it were) so that the parent string can perform the wrapping more securely if the text happens to be used in a UI context.

Reply:
> If I’m not mistaken, these changes include reverting some of the tagged strings to plain text. […] handling mixed-language strings

That shouldn't be the case. It could be that I accidentally did that in a place or two, in which case I would be happy to fix those.
There were a few cases of where a translated text is nested in another translated text as a replacement value (like [here](https://github.com/openstreetmap/iD/blob/68098489afafea9f35fdfd339948c28224e29d71/modules/ui/sections/changes.js#L32)). You are right that those could technically be mixed languages. I changed these now to be formatted as nested localized-text spans. Or do you think that these nested structure (e.g. something like `<span lang="it">Ciao, <span lang="en">mapper</span>!</span>`) could be an issue for accessibility?

Reply:
Oh great, that nested structure would be ideal. 👍 Glad I was mistaken.